### DISCUSSION TITLE
Routing: No module named functools

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#448] [Q&A]
Hi All,

Relating to [this previous post](https://anvil.works/forum/t/fixed-no-package-functool-when-importing-anvil-extras-routing/12330) about the same issue, we saw a whole bunch of logs come through last night with this same error:
<img width="496" alt="image" src="https://github.com/anvilistas/anvil-extras/assets/31876600/f9241022-03ee-4061-b408-2c4419507538">

Here is the print statements of all the user agent info:
```
Browser Info: Mozilla/5.0 (iPhone; CPU iPhone OS 16_6 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) GSA/278.0.557984068 Mobile/20G75 Safari/604.1
```

We haven't seen anything reoccur but given the very close timing of all the issues I am guessing it was a Mozilla related transient issue.

**Comment by s-cork** [comment]
When you say it didn't recur. You mean the errors were there and now they're not there any more. During what period of time did you see these errors?

**Comment by rhurlbatt** [comment]
As per the logs, the error didn't reoccur right away and we didn't see any more errors come through the logs until later that day. All the timing seemed to be within a few seconds of each other around 14:46 GMT

We also see some of this reoccuring later in the day and then again today.
It still looks like it is isolated to iphones and the Anvil Fix yesterday is definitely in effect on these devices.

**Reply by s-cork** [reply]
Do those devices always get this error or only sometimes?

**Comment by rhurlbatt** [comment]
Its not something I remember seeing. We haven't had anyone reporting this recently, its only since we added some new users and they may be picking up on things while others become desensitized to it (?)

I would have thought that we would have had reports about it over the passed 12 months if it was a consistent thing.

I have tried going through the logs but quite often we don't get them because they are offline.
